src/pages/workspaces/workspace/Dashboard.js

```diff
@@ -122,9 +123,71 @@ const RightBoxSection = ({ title, info, initialOpenState, onClick, children }) =
   ])
 }
 
+const BucketLocation = requesterPaysWrapper({ onDismiss: _.noop })(({ workspace }) => {
+  const isGoogleWorkspace = !!workspace?.workspace?.googleProject
+
+  const [loading, setLoading] = useState(true)
+  const [{ location, locationType }, setBucketLocation] = useState({ location: undefined, locationType: undefined })
+  const [needsRequesterPaysProject, setNeedsRequesterPaysProject] = useState(false)
+  const [showRequesterPaysModal, setShowRequesterPaysModal] = useState(false)
+
+  const signal = useCancellation()
+  const loadBucketLocation = useCallback(async () => {
+    setLoading(true)
+    try {
+      const { namespace, name, workspace: { googleProject, bucketName } } = workspace
+      const response = await Ajax(signal).Workspaces.workspace(namespace, name).checkBucketLocation(googleProject, bucketName)
+      setBucketLocation(response)
+    } catch (error) {
+      if (error.requesterPaysError) {
+        setNeedsRequesterPaysProject(true)
+      } else {
+        reportError('Unable to get bucket location.', error)
+      }
+    } finally {
+      setLoading(false)
+    }
+  }, [workspace, signal])
+
+  useEffect(() => {
+    if (isGoogleWorkspace) {
+      loadBucketLocation()
+    }
+  }, [isGoogleWorkspace, loadBucketLocation])
+
+  if (!isGoogleWorkspace) {
+    return null
+  }
+
+  if (loading) {
+    return 'Loading'
+  }
+
+  if (!location) {
+    return h(Fragment, [
+      'Unknown',
+      needsRequesterPaysProject && h(ButtonSecondary, {
+        tooltip: 'This workspace\'s bucket is requester pays. Click to choose a workspace to bill requests to and get the bucket\'s location.',
```

Comment:
I would suggest adding an `aria-label` with a more concise description of the button. The tooltip text will still be read out at the end, but it makes it easier to understand what the control is. Currently the screen reader announces: **"This workspace's bucket is request pays. Click at chooose….." button**.

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/3045/commits/3b7e5abb0584badeb601259230455db7856cb433